middleware/replacer.go

```diff
@@ -117,3 +123,14 @@ const (
 	timeFormat     = "02/Jan/2006:15:04:05 -0700"
 	headerReplacer = "{>"
 )
+
+var (
+	customReplacements      = map[string]string{}
+	customReplacementsMutex sync.RWMutex
+)
+
+func RegisterReplacement(name, value string) {
+	customReplacementsMutex.Lock()
+	defer customReplacementsMutex.Unlock()
+	customReplacements[name] = value
```

Comment:
So is there really a race condition without the mutex? Does your add-on call RegisterReplacement during request-time or only at the initial setup phase? (Mostly interested in the necessity of the mutex for performance reasons.)